src/ui/components/SecondaryToolbox/react-devtools/injectReactDevtoolsBackend.ts

```diff
@@ -111,5 +142,159 @@ export async function injectReactDevtoolsBackend(
   if (!pauseId) {
     return;
   }
-  await pauseEvaluationsCache.readAsync(replayClient, pauseId, null, expression);
+  await pauseEvaluationsCache.readAsync(replayClient, pauseId, null, rdtInjectionExpression);
+}
+
+function collectElementIDs(
+  store: StoreWithInternals,
+  elementID: number,
+  elementIDs: number[] = []
+) {
+  elementIDs.push(elementID);
+  const element = store._idToElement.get(elementID);
+  for (const childID of element!.children) {
+    collectElementIDs(store, childID, elementIDs);
+  }
+  return elementIDs;
 }
+
+// Evaluated in the paused browser.
+// Encapsulates the logic for finding the DOM nodes associated with each fiber ID.
+// Returns a different structured result based on object ID lookup support.
+function getComponentSpecificNodesToFiberIDs(
+  rendererIdsToFiberIds: Record<number, number[]>,
+  isObjectIdCapable: boolean
+): Map<HTMLElement, number> | ChunksArray {
+  // Modern: if we have object ID lookup in evals, save just the IDs
+  // This only works with Chromium 2023-09-16+
+  const nodeIdsToFiberIds: Record<string, number> = {};
+  // Legacy: if we don't have object ID lookup in evals, save the full objects
+  // This works with Firefox and older Chromium
+  const domNodesToFiberIds = new Map<HTMLElement, number>();
+
+  const NO_NODES: HTMLElement[] = [];
```

Comment:
nit: There isn't much value in defining a "constant" like this within the body of a function.